Library/brew.sh

```diff
@@ -209,10 +209,13 @@ EOS
   esac
 fi
 
-# Hide shellcheck complaint:
-# shellcheck source=/dev/null
-source "$HOMEBREW_LIBRARY/Homebrew/utils/ruby.sh"
-setup-ruby-path
+if [[ -z "$HOMEBREW_BASH_COMMAND" ]]
+then
+  # Hide shellcheck complaint:
+  # shellcheck source=/dev/null
+  source "$HOMEBREW_LIBRARY/Homebrew/utils/ruby.sh"
+  setup-ruby-path
+fi
```

Comment:
Would you mind to wait for a moment? As I think porting to bash is not too difficult and much better solution.